proposals/NNNN-android-manager-apis.md

```diff
@@ -0,0 +1,689 @@
+
+```java
+public class ProxyBridge implements IProxyListenerBase{
+
+	public interface OnRPCListener {
+		void onRpcReceived(int functionID, RPCMessage message);
+	}
+
+	protected interface LifecycleListener{
+		void onProxyClosed(String info, Exception e, SdlDisconnectedReason reason);
+		void onServiceEnded(OnServiceEnded serviceEnded);
+		void onServiceNACKed(OnServiceNACKed serviceNACKed);
+		void onError(String info, Exception e);
+	}
+
+	public ProxyBridge( LifecycleListener lifecycleListener){}
+
+	public boolean onRPCReceived(final RPCMessage message){}
+
+	protected void addRpcListener(FunctionID id, OnRPCListener listener){}
+
+	public boolean removeOnRPCListener(FunctionID id, OnRPCListener listener){}
+	
+	.....
+	
+	// All overridden IProxyListenerBase methods
+```
+
+#### SDLFile
+
+SDLFile will be added to further match the API with iOS. It will contain helpers that do some of the work that we required the developer to do before, like automatically setting the file type and assembling the byte array of the file to be sent. For uploading a file, the developer may choose to pass either a resource id, a URI, or the byte array itself. 
+
+
+```java
+public SdlFile{
+
+	public void setName(@NonNull String fileName) 
+	public String getName()
+
+	public void setResourceId(int id, Resources resources)
+	public void getResourceId()
+
+	public void setUri(URI uri)
+	public byte[] getUri()
+
+	public void setFileData(byte[] data)
+	public byte[] getFileData()
+
+	public void setType(@NonNull FileType fileType)
+	public FileType getType()
+
+	public void setPersistent(boolean persistentFile)
+	public boolean isPersistent()
+}
+```
+
+###### SDLArtwork
+
+`SDLArtwork` will extend `SDLFile` and be specifically for uploading images.
+
+
+#### CompletionListener
+
+A simple listener with a single method, `onComplete(boolean success)` will be added. It's purpose is simply understand when an asynchronous call int he library was completed, and if it was completed successfully or not. 
+
+```java 
+public interface CompletionListener {
+	/**
+	 * Returns whether a specific operation was successful or not
+	 * @param success - success or fail
+	 */
+	void onComplete(boolean success);
+}
+```
+
+### Future Manager Alignment
+
+Future managers, like the recently approved [Menu Manager](https://github.com/smartdevicelink/sdl_evolution/pull/440), and [Choice Set Manager](https://github.com/smartdevicelink/sdl_evolution/pull/453) will now be able to be easily added with this system in place. This means we should no longer experience a large divergence between the iOS and Android library in terms of developer facing APIs.
```

Comment:
The links here should reference the proposal files: https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0155-mobile-menu-manager.md and https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0157-mobile-choice-manager.md.